lib/handlers/allow.js

```diff
@@ -43,6 +43,10 @@ function allow (mode, checkPermissionsForDirectory) {
       resourcePath += '/'
     }
 
+    let trustedOrigins = [ldp.resourceMapper.resolveUrl(req.hostname)].concat(ldp.trustedOrigins)
```

Comment:
An even more readable version would be to have it in separate function; but this is good ^_^